src/main/java/repository/CourseMongoRepository.java

```diff
@@ -0,0 +1,80 @@
+package repository;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.bson.Document;
+
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+
+import model.Course;
+
+public class CourseMongoRepository implements CourseRepository{
+	
+	private static final String ID = "id";
+	private static final String STUDENTS = "students";
+	private MongoCollection<Document> courseCollection;
+
+	public CourseMongoRepository(MongoClient mongoClient, String dbName, String dbCollection) {
+		courseCollection = mongoClient.getDatabase(dbName).getCollection(dbCollection);
+	}
+
+	@Override
+	public List<Course> findAll() {
+		return StreamSupport.
+				stream(courseCollection.find().spliterator(), false)
+				.map(this::fromDocumentToCourse)
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public Course findById(String id) {
+		Document document  = courseCollection.find(Filters.eq(ID, id)).first();
+		if(document != null) {
```

Comment:
I don't know if this is a Code Smell, but we could change it simply by returning the result of `fromDocumentToCourse`

Reply:
We could but we have to test it: this code is exactly like the code on the book

Reply:
Ok, if this is not a code smell, we can skip this change.

Reply:
It's a code smell indeed. We can choose to either change the method (somehow) or supress the code smell with a good motivation